src/plans/stopping_criterion.jl

```diff
@@ -43,6 +43,24 @@ function update_stopping_criterion!(c::StopAfterIteration, ::Val{:MaxIteration},
     return c
 end
 
+"""
+    StopWhenTimeElapsed <: StoppingCriterion
+
+stores a threshold indicating when to stop after some elapsed time  [`GradientProblem`](@ref).
+"""
+mutable struct StopWhenTimeElapsed <: StoppingCriterion
```

Comment:
Instead of this you can just use

https://manoptjl.org/stable/plans/stopping_criteria/#Manopt.StopAfter

which does the same (and stores the time internally instead of requiring _every_ option to have a `timer` field, which only just one option has. As mentioned in the comment above, I would prefer having a StepTimer debug/record as well similar to the timer stopping criterion (instead of the timer field in options - which would have to be reimplemented in every options).